File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        
        map<int, int> my_map;
        map<int, int>::iterator iter;

        for(int i = 0; i< nums.size() ; i++){

            iter = my_map.find(nums[i]);

            if(iter != my_map.end()){
                if( abs(iter->second - i) <= k)
                    return true;

                iter->second = i;
            }
            else{
                my_map[nums[i]] = i;
            }
        }
        return false;

    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
         # sliding window keep the window size <= k
        left = 0
        nums_in_window = defaultdict(int)

        for right in range(len(nums)):
            nums_in_window[nums[right]] += 1

            while right - left > k:
                nums_in_window[nums[left]] -= 1
                left += 1

            if nums_in_window[nums[right]] == 2:
                return True  
        
        return False

        

        